harmony/harmony.py

```diff
@@ -180,6 +196,12 @@ def __init__(self,
             (lambda tr: tr['start'] < tr['stop'] if 'start' in tr and 'stop' in tr else True,
              'The temporal range\'s start must be earlier than its stop datetime.')
         ]
+        self.shape_validations = [
+            (lambda s: os.path.isfile(s), 'The provided shape path is not a file'),
+            (lambda s: s.split('.').pop().lower() in _shapefile_exts_to_mimes,
+             'The provided shape file is not a recognized type.  Valid file extensions: '
```

Comment:
I know this approach to validation is consistent with what is done for spatial/temporal validation, but it has the drawback of not being able to include the provided type in the error message. Error messages like "You didn't provide a valid X. Valid Xs are ..." have always been a source of annoyance for me. I prefer "You provided 'foo' which is not a valid X. Valid Xs are ...".